src/picologging/handlers.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+import picologging
+
+class BaseRotatingHandler(picologging.FileHandler):
+    """
+    Base class for handlers that rotate log files at a certain point.
+    Not meant to be instantiated directly.  Instead, use RotatingFileHandler
+    or TimedRotatingFileHandler.
+    """
+    namer = None
+    rotator = None
+
+    def __init__(self, filename, mode, encoding=None, delay=False):
+        """
+        Use the specified filename for streamed logging
+        """
+        picologging.FileHandler.__init__(self, filename, mode=mode,
+                                     encoding=encoding, delay=delay)
+        self.mode = mode
+        self.encoding = encoding
+
+    def shouldRollover(self, record):
+        """
+        Determine if rollover should occur.
+
+        Should be implemented in inherited classes.
+        """
+        raise NotImplementedError()
+
+    def doRollover(self, record):
+        """
+        Do a rollover.
+
+        Should be implemented in inherited classes.
+        """
+        raise NotImplementedError()
+
+    def emit(self, record):
+        """
+        Emit a record.
+
+        Output the record to the file, catering for rollover as described
+        in doRollover().
+        """
+        try:
+            if self.shouldRollover(record):
+                self.doRollover()
+            picologging.FileHandler.emit(self, record)
```

Comment:
Just like in the other PR, this isn't the right way to call parent type methods, I don't understand why the standard library does it like this. 

It should use `super()` and the parent should be initialized correctly. You can be specific in super calls to the parent method to invoke.